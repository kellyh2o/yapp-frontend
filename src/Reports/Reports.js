import React from 'react';
import { connect } from "react-redux";
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Title from '../Dashboard/Title';
import moment from 'moment';
import LocationHeader from '../Dashboard/LocationHeader';


const useStyles = makeStyles((theme) => ({  
  paper: {
    padding: theme.spacing(2),
    display: 'flex',
    overflow: 'auto',
    flexDirection: 'column',
  },
  fixedHeight: {
    height: 250,
  },
}));

function buildTableData(jump) {

  let tableData = [];

  jump.events.forEach(jumpEvent => {

      let customId = `${jump._id}-----${jumpEvent._id}`;

      if (tableData.filter(data => data.id === customId).length === 0) {
          tableData.push({
              id: customId,
              date: new Date(jumpEvent.timeStamp),
              jump: jump.name,
              riderMass: jumpEvent.riderMass,
              riderSpeed: jumpEvent.riderSpeed
          });
      }
  });

  return tableData.sort((x, y) => {
      return y.date - x.date;
  });
}

const Reports = (props) => {
  const classes = useStyles();

  return (
    <div>
      <Paper style={{marginBottom: "20px", padding: "20px"}}>
        <LocationHeader />
      </Paper>
      <Grid container spacing={3}>
        {/* Jump Event Tables */}
        {props.demoLocation && props.demoLocation.jumps ? 
          props.demoLocation.jumps.map(jump => {
            let tableData = buildTableData(jump);
            return (
              <Grid key={jump._id} item xs={12}>
                <Paper className={classes.paper}>
                  <React.Fragment>
                    <Title>{jump.name} Jump Events</Title>
                    <Table size="small">
                      <TableHead>
                        <TableRow>
                         <TableCell>Date</TableCell>
                         <TableCell>Rider Mass (lbs) </TableCell>
                         <TableCell>Rider Speed (mph) </TableCell>
                        </TableRow>
                      </TableHead>
                      <TableBody>
                        {tableData.map(data => (
                          <TableRow key={data.id}>
                            <TableCell>{moment(data.date).calendar()}</TableCell>
                            <TableCell>{data.riderMass}</TableCell>
                            <TableCell>{data.riderSpeed}</TableCell>
                          </TableRow>
                        ))}
                      </TableBody>
                    </Table>
                  </React.Fragment>
                </Paper>
              </Grid>
            );
          })
        :
        (<div />)
        }
      </Grid>
    </div>
  );
}
const mapStateToProps = (state) => {
  return {
      demoLocation: state.demoLocation
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
      selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Reports);