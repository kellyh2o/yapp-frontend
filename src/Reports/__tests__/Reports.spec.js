import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Reports from "../Reports";

describe('Reports', () => {
    it("should render", () => {
        const mockStore = configureMockStore();

        const initialState = {
            demoLocation: {
                name: "Hogwarts",
                jumps: [ 
                    {
                        _id: "1",
                        name: "Jump1",
                        events: []
                    },
                    {
                        _id: "2",
                        name: "Jump2",
                        events: []
                    },
                    {
                        _id: "3",
                        name: "Jump3",
                        events: [
                            {
                                _id: "3a",
                                riderMass: 22,
                                riderSpeed: 12,
                            }
                        ]
                    }
                ]
            }
        };
        const store = mockStore(initialState);

        const { asFragment } = render(
            <Provider store={store}>
                <Reports />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    })
});

