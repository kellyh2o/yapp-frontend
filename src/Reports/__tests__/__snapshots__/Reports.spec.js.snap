// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports should render 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-bottom: 20px; padding: 20px;"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        Location
      </h2>
      <p
        class="MuiTypography-root MuiTypography-h4"
        style="padding-top: 20px;"
      >
        Hogwarts
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            Jump1 Jump Events
          </h2>
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Mass (lbs) 
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Speed (mph) 
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            />
          </table>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            Jump2 Jump Events
          </h2>
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Mass (lbs) 
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Speed (mph) 
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            />
          </table>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            Jump3 Jump Events
          </h2>
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Mass (lbs) 
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Speed (mph) 
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  Invalid date
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  22
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  12
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
